eclass/freebsd.eclass

```diff
@@ -95,6 +95,23 @@ freebsd_get_bmake() {
 	echo "${bmake}"
 }
 
+# Generates a patch SRC_URI or DISTDIR of upstream.
+freebsd_upstream_patches() {
+	local opt=$1
+	[[ ${#UPSTREAM_PATCHES[@]} -eq 0 ]] && return 1
+	for x in ${UPSTREAM_PATCHES[@]}
+	do
+		local out=${x/\//-}
+		out=${out/:/}
+		out="${PN}-${RV}-${out}"
```

Comment:
I think it would be easier to start with:

    local out=${PN}-${RV}-${x/\//-}

(unless PN/RV contains `:`)